go-controller/pkg/ovn/ovn.go

```diff
@@ -780,6 +780,21 @@ func (oc *Controller) WatchNetworkPolicy() {
 		AddFunc: func(obj interface{}) {
 			policy := obj.(*kapisnetworking.NetworkPolicy)
 			oc.initRetryPolicy(policy)
+			oc.checkAndSkipRetryPolicy(policy)
+			// If there is a delete entry this is a network policy being added
+			// with the same name as a previous network policy that failed deletion.
+			// Destroy it first before we add the new policy.
+			if oc.retryEntryHasDelete(policy) {
+				klog.Infof("Detected stale policy during new policy add with the same name: %s/%s",
+					policy.Namespace, policy.Name)
+				retryEntry := oc.getRetryEntry(policy)
+				if err := oc.deleteNetworkPolicy(retryEntry.oldPolicy, nil); err != nil {
```

Comment:
hmm so the delete should already be in the retry loop for that old one right? so should we simply just return if the entry is in retry cache? do we need to re-trigger the delete here? that way we don't have to skip the retry just to unskip it here again....

Reply:
At this point we want to add the new network policy now, and not wait for a retry handler to do it later. So we do our best  to configure it ASAP, else let the retry handler do it later.